### ISSUE TITLE
discuss possibility to reborn CODEOWNERS thanks to the great feature to notify the relevant team when a PR arrives

### ISSUE THREAD

**Issue by valpinkman** [LedgerHQ/ledger-live#514]
after a discussion with Hakim, we came to conclusion that CODEOWNERS were solving a great problem in the past: getting notified about code that is relevant to you.

this ticket is created to have the future opportunity to re consider this feature we used to have.

DISCUSSION NEEDED WITH THE TEAM

**Comment by valpinkman** [comment]
DISCUSSION NEEDED WITH THE TEAM

**Comment by chabroA** [comment]
also this, related (and could be part of the solution of the problem) -> https://docs.github.com/en/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team#about-auto-assignment

When assigning a team, it would be more efficient to have a dedicated reviewer within the team for this PR. Avoids the "someone else will review" fallacy

**Comment by lambertkevin** [comment]
Codeowners are spam generating, and they are assuming everybody in a team is knowledgable on all subjects related to the said team, which is wrong for 90% of us I think.

@LFBarreto suggested to use the "suggested reviewers" feature which seems like a good idea to me for internal PRs.

For exeternal contributors, we could have either instructions in the CONTRIBUTING.md about who it should be assigned to depending on the feature, or have a dedicated group of people being assigned with the PR (review relayers in a way) and that are in charge to reassign it to the tech lead in charge.

**Comment by haammar-ledger** [comment]
I think PRs assigned to the `ledger-live` group was mostly the issue in the past, because you could be assigned to PRs where you really had 0 context and relevance.

But while all PRs from Live may not be of our concern, IMHO all PRs from our squad should be. I don't consider as spam to see notifications of every PR from people in my squad. The "who does the review" is solved simply if everyone just takes a small time buffer to review what's not reviewed yet. Then PRs get reviewed one after another, with a bit less friction than being assigned one by one.

**Comment by gre** [comment]
@lambertkevin if you are part of a team that own code, you must be informed and must look at what's happening in your scope. always having the same suggested reviewers is not very inclusive.
then it's all about maintaining your Github notification inbox, it is a bit spammy at first, but you can configure it so it's a daily routine and there is not that much PR these days. we're only at 50 PR merged per week, back in 2020 we were around 80-100 PR per week.

**Comment by valpinkman** [comment]
From my point of view, I agree with @haammar-ledger that `ledger-live` shouldn't be a codeowner on anything but we can definitely target the right squad on specific part of the code. 
I would say we  again need to define which squads would benefit the most from this and set it up basically.

Also it'd be interesting to look into the autoassignment proposed by @chabroA

**Comment by lambertkevin** [comment]
@gre Yes I want to be informed. I don't want to receive an email every single time someone creates a PR modify a file I'm supposed to be responsible for, add a commit, a comment, is merging, is being reviewed by someone else etc etc etc... Although that's true on someone else's PR, not necessarily on mine, I want to know everything about my PRs. 
Problem is, every time *we* tried to configure it, it was either not receiving anything about your PRs or receiving everything about everything.

Lastly, saying that everyone in a team should be knowledgeable enough to be notified about anything related to the said team is not very realistic. I wish it was, but that's not how our "squads/teams" are working 😞

**Comment by gre** [comment]
@lambertkevin yeah you are right, blockchain support team is an example of this not working well esp if we have tons of different contributions
